fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -409,7 +446,18 @@ ACTOR Future<Void> simulatedMachine(ClusterConnectionString connStr, std::vector
 				Reference<ClusterConnectionFile> clusterFile(useSeedFile ? new ClusterConnectionFile(path, connStr.toString()) : new ClusterConnectionFile(path));
 				const int listenPort = i*listenPerProcess + 1;
 				AgentMode agentMode = runBackupAgents == AgentOnly ? ( i == ips.size()-1 ? AgentOnly : AgentNone ) : runBackupAgents;
-				processes.push_back(simulatedFDBDRebooter(clusterFile, ips[i], sslEnabled, tlsOptions, listenPort, listenPerProcess, localities, processClass, &myFolders[i], &coordFolders[i], baseFolder, connStr, useSeedFile, agentMode, whitelistBinPaths));
+				if (processClass == ProcessClass::StorageClass) {
+					// Test if locality is misconfigured for storage process
+					processes.push_back(simulatedInvalidLocalityFDBDRebooter(
```

Comment:
It looks like this is going to do locality misconfiguring fairly frequently. I think it would probably be good to have a little bit wider distribution of behaviors, such as tests that don't misconfigure locality at all, some that do it a lot, and some that only do it a little. Also, is there any reason not to test locality misconfigurations on other process types besides storage?

Reply:
One other thing -- I'm not that well-versed in how the simulated killing stuff works, but we should be careful that we don't leave a process in a state with misconfigured locality when the simulator thinks that process is running. For example, I'm imagining a case where the cluster reboots a process (such that it starts back up) and considers it running normally, but where we've now put it in a state where it's not functioning in the cluster. If the simulator then thinks it can kill a bunch more processes, we might end up in an invalid state.

Reply:
Yes, you are right! In case that situation happens, we will see failures in correctness, say `ConsistencyCheck` fails because the misconfigured process doesn't to have a storage server on it.